project_task_analytic_propagation/models/project_task.py

```diff
@@ -1,20 +1,37 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/LGPL-3.0)
 from odoo import api, fields, models
 
 
 class ProjectTask(models.Model):
-    _inherit = "project.task"
+    _name = "project.task"
+    _inherit = ["project.task", "analytic.plan.fields.mixin"]
 
-    analytic_account_id = fields.Many2one(
-        "account.analytic.account",
-        ondelete="set null",
-        compute="_compute_analytic_account_id",
-        store=True,
+    account_id = fields.Many2one(
+        compute="_compute_task_analytic_accounts",
         readonly=False,
-        domain="[('company_id', '=?', company_id)]",
-        help="Analytic account to which this task and its timesheets are linked.",
+        store=True,
     )
 
-    @api.depends("project_id.account_id")
-    def _compute_analytic_account_id(self):
+    def _get_project_analytic_plan_columns(self):
+        return [
+            f"project_id.{x}"
+            for x in self.project_id._get_plan_fnames()
+            if self._fields.get(x)
+        ]
+
+    @api.depends(lambda self: self._get_project_analytic_plan_columns())
+    def _compute_task_analytic_accounts(self):
+        plan_fnames = self._get_plan_fnames()
         for task in self:
-            task.analytic_account_id = task.project_id.account_id
+            vals_to_update = {}
+            for fname in plan_fnames:
+                if task._fields.get(fname) and not task[fname]:
+                    vals_to_update[fname] = task.project_id[fname]
+            if vals_to_update:
+                task.write(vals_to_update)
```

Comment:
issue: if you update a plan account in the project, it won't be updated in the task if it already had one.

Your current code is only updating task accounts if they were empty, but that's not probably what we want.

Besides, code can be simplified just like in the suggestion above.

```suggestion
        for task in self:
            for fname in plan_fnames:
                if task.project_id[fname]:
                    task[fname] = task.project_id[fname]
```